### PR TITLE
add support for windows

### LINKED ISSUE #29: Windows support

### PR DESCRIPTION
**Pull request by aramase**
Fixes #29

### CODE REVIEW

File: deployment/provider-azure-installer-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      labels:
+        app: csi-secrets-store-provider-azure
+    spec:
+      nodeSelector:
+        beta.kubernetes.io/os: windows
+      containers:
+        - name: provider-azure-installer
+          image: aramase/secrets-store-csi-driver-provider-azure:win00
+          imagePullPolicy: Always
+          # resources:
```
Comment (ritazh) [inline deployment/provider-azure-installer-windows.yaml]: why are these commented out?

File: provider.go [hunk 2]
```diff
@@ -363,7 +363,7 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 			return err
 		}
 		objectContent := []byte(content)
-		if err := ioutil.WriteFile(path.Join(targetPath, keyVaultObject.ObjectName), objectContent, permission); err != nil {
+		if err := ioutil.WriteFile(filepath.Join(targetPath, keyVaultObject.ObjectName), objectContent, permission); err != nil {
```
Comment (ritazh) [inline provider.go]: 👍

File: windows.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FROM mcr.microsoft.com/windows/servercore:1809 as core
+
+FROM mcr.microsoft.com/windows/servercore:ltsc2019
+LABEL description="Secrets Store CSI Driver Provider Azure"
+ARG IMAGE_VERSION=0.0.3
```
Comment (ritazh) [inline windows.Dockerfile]: nit: the next tag is probably `0.0.4`?

**Comment by ritazh** [comment]
Just few nits but mostly looks great. Can we also add e2e to test this with aks-engine in a hybrid cluster?

File: windows.Dockerfile (hunk 3 above)
Comment (aramase) [reply windows.Dockerfile]: Updated!

File: deployment/provider-azure-installer-windows.yaml (hunk 1 above)
Comment (aramase) [reply deployment/provider-azure-installer-windows.yaml]: Updated the limits!

**Comment by aramase** [comment]
@ritazh This PR is ready for review and merge

File: deployment/provider-azure-installer-windows.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  template:
+    metadata:
+      labels:
+        app: csi-secrets-store-provider-azure
+    spec:
+      nodeSelector:
+        beta.kubernetes.io/os: windows
+      containers:
+        - name: provider-azure-installer
+          image: aramase/secrets-store-csi-driver-provider-azure:win00
```
Comment (ritazh) [inline deployment/provider-azure-installer-windows.yaml]: pls create an issue to update this after we cut a new release

**Comment by ritazh** [comment]
FYI, I have added https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/37 to keep track of signing of these new windows binaries.

File: deployment/provider-azure-installer-windows.yaml (hunk 4 above)
Comment (aramase) [reply deployment/provider-azure-installer-windows.yaml]: Opened an issue to track updating image - https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/38

**Pull request merged**